lib/checks/canFindDenominatorInNumerator.js

```diff
@@ -19,17 +20,42 @@ function canFindDenominatorInNumerator(node) {
   if (Node.Type.isParenthesis(denominator)) {
     denominator = denominator.content;
   }
+  if (!(numerator.op === '+' || numerator.op === '-' ||
+        denominator.op === '+' || numerator.op === '-')) {
+    return false;
+  }
+  if (denominator.op !== '+') {
+    return false;
+  }
+
+  let numeratorFirstTerm;
+  if (numerator.op === '+') {
+    numeratorFirstTerm = new Node.PolynomialTerm(numerator.args[0]);
+  }
+  else if (numerator.op === '*') {
+    numeratorFirstTerm = new Node.PolynomialTerm(numerator);
+  }
+
+  let denominatorFirstTerm;
+  if (denominator.op === '+') {
+    denominatorFirstTerm = new Node.PolynomialTerm(denominator.args[0]);
+  }
+  else if (denominator.op === '*') {
+    denominatorFirstTerm = new Node.PolynomialTerm(denominator);
+  }
 
-  if (numerator.op !== '+') {
+  if (!(numeratorFirstTerm)) {
     return false;
   }
-  if (!('args' in denominator)) {
+  if (!(denominatorFirstTerm)) {
+    return false;
+  }
+
+  if (!(numeratorFirstTerm.getSymbolName() === 'x' && denominatorFirstTerm.getSymbolName() === 'x')) {
```

Comment:
I think it's better to just say they have the same symbol. Then if they share the symbol `y`, it'd work too!